src/PackageProject.cmake

```diff
@@ -0,0 +1,157 @@
+# Uses ycm (permissive BSD-3-Clause license) and FowardArguments (permissive MIT license)
+
+function(package_project)
+  cmake_policy(SET CMP0103 NEW) # disallow multiple calls with the same NAME
+
+  set(_options ARCH_INDEPENDENT # default to false
+  )
+  set(_oneValueArgs
+      # default to the project_name:
+      NAME
+      COMPONENT
+      # default to project version:
+      VERSION
+      # default to any newer:
+      COMPATIBILITY
+      # default to ${CMAKE_BINARY_DIR}
+      CONFIG_EXPORT_DESTINATION
+      # default to ${CMAKE_INSTALL_DATADIR}/cmake/${NAME} suitable for vcpkg, etc.
+      CONFIG_INSTALL_DESTINATION)
+  set(_multiValueArgs
+      # recursively found for the current folder if not specified
+      TARGETS
+      # a list of public/interface include directories or files
+      PUBLIC_INCLUDES
+      # the names of the INTERFACE/PUBLIC dependencies that are found using `CONFIG`
+      PUBLIC_DEPENDENCIES_CONFIGED
```

Comment:
Should this be named `PUBLIC_DEPENDENCIES_BY_CONFIG` instead?

Reply:
https://github.com/cpp-best-practices/project_options/pull/58/commits/816fd6b89899a759fae0392af111592dbbfe758b